src/Symfony/Component/HttpKernel/Profiler/FileProfilerStorage.php

```diff
@@ -42,7 +44,9 @@ public function __construct($dsn)
         $this->folder = substr($dsn, 5);
 
         if (!is_dir($this->folder)) {
-            mkdir($this->folder, 0777, true);
+            if (false === @mkdir($this->folder, 0777, true) && !is_dir($this->folder)) {
+                throw new IOException(sprintf('Unable to create the storage directory (%s).', $this->folder));
```

Comment:
The HttpKernel should not need to know the Filesystem Component exceptions. Simply use `\RuntimeException`